src/MakePotCommand.php

```diff
@@ -106,29 +108,34 @@ public function __invoke( $args, $assoc_args ) {
 
 		$ignore_domain = Utils\get_flag_value( $assoc_args, 'ignore-domain', false );
 
-		if ( ! $ignore_domain ) {
-			$this->domain = Utils\get_flag_value( $assoc_args, 'domain', $this->slug );
-		}
-
 		if ( ! $this->source || ! is_dir( $this->source ) ) {
 			WP_CLI::error( 'Not a valid source directory!' );
 		}
 
 		$this->retrieve_main_file_data();
 
-		// Determine destination.
-		if ( isset( $args[1] ) ) {
-			$this->destination = $args[1];
-		} else {
-			$file_data = $this->get_main_file_data();
+		$file_data = $this->get_main_file_data();
 
-			// Current directory.
-			$this->destination = $this->slug . '.pot';
+		if ( ! $ignore_domain ) {
+			$this->domain = $this->slug;
 
-			if ( isset( $file_data['Domain Path'] ) ) {
-				// Domain Path inside source folder.
-				$this->destination = $this->source . DIRECTORY_SEPARATOR . $file_data['Domain Path'] . DIRECTORY_SEPARATOR . $this->slug . '.pot';
+			if ( ! empty( $file_data['Text Domain'] ) ) {
+				$this->domain = $file_data['Text Domain'];
 			}
+
+			$this->domain = Utils\get_flag_value( $assoc_args, 'domain', $this->domain );
+		}
+
+		// Determine destination.
+		$this->destination = $this->source . DIRECTORY_SEPARATOR . $this->slug . '.pot';
```

Comment:
The use of `DIRECTORY_SEPARATOR` is actually not a good practice here. That constant should be sued for functions like `explode()` or `(r|l)trim`. If you compose a filesystem string for the server, just use `/` instead.

So, I'd write this line as follows:
```
$this->destination = "{$this->source}/{$this->slug}.pot";
```

Reply:
Thanks, that makes sense!